vmdb/app/models/ems_kubernetes.rb

```diff
@@ -36,27 +52,40 @@ def api_endpoint
     self.class.raw_api_endpoint(hostname, port)
   end
 
-  def connect(_options = {})
-    self.class.raw_connect(hostname, port)
-  end
+  def connect(options = {})
+    hostname = options[:hostname] || self.address
+    port     = options[:port]     || self.port
 
-  def self.event_monitor_class
-    MiqEventCatcherKubernetes
+    self.class.raw_connect(hostname, port)
   end
 
-  def authentication_check
+  def verify_credentials(auth_type = nil, options = {})
     # TODO: support real authentication using certificates
-    [true, ""]
-  end
+    options = options.merge(:auth_type => auth_type)
 
-  def verify_credentials(_auth_type = nil, _options = {})
-    # TODO: support real authentication using certificates
-    true
+    with_provider_connection(options) do |connection|
+      unless (connection.api['versions'] || []).include? api_version
```

Comment:
@Fryguy kubeclient supports all the APIs, it's ManageIQ being picky on which one supports, so the validation should be here. For as much as the `api_valid?` method is involved... the api would be valid but ManageIQ wouldn't be able to parse the objects.